FirebaseRemoteConfig/Sources/RCNConfigContent.m

```diff
@@ -469,8 +471,50 @@ - (FIRRemoteConfigUpdate *)getConfigUpdateForNamespace:(NSString *)FIRNamespace
     }
   }
 
+  NSMutableDictionary<NSString *, NSMutableDictionary *> *fetchedRollout =
+      [self getParameterKeyToRolloutMetadataDic:fetchedRolloutMetadata];
+  NSMutableDictionary<NSString *, NSMutableDictionary *> *activeRollout =
+      [self getParameterKeyToRolloutMetadataDic:activeRolloutMetadata];
+
+  // add params with new/updated rollout metadata
+  for (NSString *key in [fetchedRollout allKeys]) {
+    if (activeRollout[key] == nil ||
+        ![activeRollout[key] isEqualToDictionary:fetchedRollout[key]]) {
+      [updatedKeys addObject:key];
+    }
+  }
+  // add params with deleted rollout metadata
+  for (NSString *key in [activeRollout allKeys]) {
+    if (fetchedRollout[key] == nil) {
+      [updatedKeys addObject:key];
+    }
+  }
+
   configUpdate = [[FIRRemoteConfigUpdate alloc] initWithUpdatedKeys:updatedKeys];
   return configUpdate;
 }
 
+- (NSMutableDictionary<NSString *, NSMutableDictionary *> *)getParameterKeyToRolloutMetadataDic:
+    (NSArray<NSDictionary *> *)rolloutMetadata {
+  NSMutableDictionary<NSString *, NSMutableDictionary *> *result =
+      [[NSMutableDictionary alloc] init];
+  for (NSDictionary *metadata in rolloutMetadata) {
+    NSString *rolloutId = metadata[RCNFetchResponseKeyRolloutID];
+    NSString *variantId = metadata[RCNFetchResponseKeyVariantID];
+    NSArray<NSString *> *affectedKeys = metadata[RCNFetchResponseKeyAffectedParameterKeys];
+    if (rolloutId && variantId && affectedKeys) {
+      for (NSString *key in affectedKeys) {
+        if (result[key]) {
+          NSMutableDictionary *rolloutIdToVariantId = result[key];
+          [rolloutIdToVariantId setValue:variantId forKey:rolloutId];
+        } else {
+          NSMutableDictionary *rolloutIdToVariantId = [@{rolloutId : variantId} mutableCopy];
```

Comment:
Could you use `copy` here instead of `mutableCopy`? And thinking more along those lines, if possible return an immutable dictionary from this method

Reply:
The rolloutIdsToVariantId dic need to be insert new pairs while doing the diff if the same key included in different rollouts. (If this case exists)
Changed the final result to be immutable and return it.